### PR TITLE
Small technical improvements / fixes + 1 new spell (eye of evil)

### PR DESCRIPTION
**Pull request by hwoarangmy**

### CODE REVIEW

File: source/entities/Creature.cpp [hunk 1]
```diff
@@ -1887,8 +1887,7 @@ void Creature::createStatsWindow()
                                         CEGUI::Event::Subscriber(&Creature::CloseStatsWindow, this));
 
     // Set the window title
-    childWindow = mStatsWindow->getChild("__auto_titlebar__");
-    childWindow->setText(getName() + " (" + getDefinition()->getClassName() + ")");
+    mStatsWindow->setText(getName() + " (" + getDefinition()->getClassName() + ")");
```
Comment (Bertram25) [inline source/entities/Creature.cpp:1891]: good!

File: source/entities/Creature.cpp [hunk 2]
```diff
@@ -1881,10 +1882,9 @@ void Creature::createStatsWindow()
     textWindow->setProperty("FrameEnabled", "False");
     textWindow->setProperty("BackgroundEnabled", "False");
 
-    // Search for the autoclose button and make it work
-    CEGUI::Window* childWindow = mStatsWindow->getChild("__auto_closebutton__");
-    childWindow->subscribeEvent(CEGUI::PushButton::EventClicked,
-                                        CEGUI::Event::Subscriber(&Creature::CloseStatsWindow, this));
+    // We want to close the window when the cross is clicked
```
Comment (Bertram25) [inline source/entities/Creature.cpp:1886]: Good riddance of the auto close button ref. :)

File: source/entities/DoorEntity.cpp [hunk 3]
```diff
@@ -49,7 +49,8 @@ DoorEntity::DoorEntity(GameMap* gameMap, Building& building, const std::string&
 }
 
 DoorEntity::DoorEntity(GameMap* gameMap) :
-    TrapEntity(gameMap)
+    TrapEntity(gameMap),
+    mBuilding(nullptr)
```
Comment (Bertram25) [inline source/entities/DoorEntity.cpp:53]: Ah, indeed.

File: source/network/ODServer.cpp [hunk 4]
```diff
@@ -191,6 +193,7 @@ bool ODServer::startServer(const std::string& creator, const std::string& levelF
         if(!MasterServer::registerGame(ODApplication::VERSION, creator, port, label, descr, uuid))
         {
             OD_LOG_ERR("Could not register the game in the master server !!!");
+            stopServer();
             return false;
```
Comment (Bertram25) [inline source/network/ODServer.cpp:197]: Looks good.

File: source/utils/StackTraceWinMinGW.cpp [hunk 5]
```diff
@@ -17,6 +17,9 @@
```
Comment (Bertram25) [inline source/utils/StackTraceWinMinGW.cpp:17]: That one looks good. Cannot really tell for the similar other ones, though,

Comment (hwoarangmy) [inline source/utils/StackTraceWinMinGW.cpp:17]: Note that I've tested for the 3 and I confirm it works

**Comment by hwoarangmy** [comment]
@OpenDungeons/developers I have been playing with blender trying rigging/animating and I found a simple model that could be good for evil eye. So, I've implemented it ^^
Note that when the evil eye is cast, the model is visible by everybody (but the tiles are visible by the caster and his allies only)

**Comment by akien-mga** [comment]
Any idea why the MinGW builds fail on AppVeyor? https://ci.appveyor.com/project/hwoarangmy/opendungeons-crxx4/build/1.0.219/job/8hc4kn80jtng0n7j
It's not really verbose, sounds like it might be failing on the 7zip command.

File: source/entities/Creature.cpp (hunk 1 above)
Comment (akien-mga) [inline source/entities/Creature.cpp:1891]: Small git-fu notice: whenever possible, try to split git commit logs into a short title + a more lengthy description after an empty newline, e.g.:

```
Change creature window stats title by using setText

Previously, it was addressing built-in window which caused a crash when using CEGUI 0.8.5.
```

That would prevent a broken long title as we have here.

File: source/render/MovableTextOverlay.cpp [hunk 6]
```diff
@@ -137,6 +137,14 @@ void ChildOverlay::update(Ogre::Real timeSincelastFrame)
     mOverlayContainer->hide();
 }
 
+void ChildOverlay::isOnScreen(bool onScreen)
```
Comment (akien-mga) [inline source/render/MovableTextOverlay.cpp:140]: Maybe `setOnScreen` would be a better name? We tend to use `isSomething` names for methods that return a boolean.

**Comment by akien-mga** [comment]
I've had a quick look at all commits, tons of good stuff! :D
Let's merge as it all seems pretty solid and @Bertram25 also tested most of it already.

File: source/render/MovableTextOverlay.cpp (hunk 6 above)
Comment (hwoarangmy) [inline source/render/MovableTextOverlay.cpp:140]: Yep, good idea

**Comment by hwoarangmy** [comment]
> Any idea why the MinGW builds fail on AppVeyor?

No. Dependencies are correctly unzipped for msvs2013 so there should be no right problem. Now that this PR is in, I will try to have a look.

**Pull request merged**